### ISSUE TITLE
Unread message count showing for muted, archived chats when using multiple accounts

### ISSUE THREAD

**Issue by Bibo-Joshi** [telegramdesktop/tdesktop#16927]
### Steps to reproduce

1. Be member in a channel, then mute & archive the channel 
2. Receive a new message in the channel
3. switch to your secondary account

### Expected behaviour

No "unread message count" is showing for the primary account

### Actual behaviour

The unread messages from the muted & archived channel are shows as unread messages:

![image](https://user-images.githubusercontent.com/22366557/132683761-31ec6757-a8d0-416a-880b-72738d6f843f.png)
![image](https://user-images.githubusercontent.com/22366557/132683887-74859a17-d9f1-4486-892e-c99b45afa323.png)

Note that:
- the official android client does not show this behavior.
- When switching back to the primary account, the unread messages count does not show

### Operating system

Windows 10, 10.0.19043 Build 19043

### Version of Telegram Desktop

3.0.1

### Installation source

Static binary from official website

**Comment by Bibo-Joshi** [comment]
> We therefore assume that the user has lost interest or resolved the problem on their own.

this is not the case. The behavior still persists in recent updates of tgdesktop.

**Comment by Bibo-Joshi** [comment]
> We therefore assume that the user has lost interest or resolved the problem on their own.

this is not the case. The behavior still persists in recent updates of tgdesktop.

**Comment by d-air1** [comment]
Can the bot not be disabled for this thread? This is just sad to look at. The only conversation on here is between a bot trying to close an issue that hasn't even been commented on by a developer and the person telling the bot that this still hasn't been fixed.

**Comment by ilya-fedin** [comment]
@d-air1 there's no enough people to look at every issue, so yes, if a person is no longer interested to confirm a bug still exists, the issue is closed to clear the tracker for issue people really care about.

**Comment by d-air1** [comment]
> @d-air1 there's no enough people to look at every issue, so yes, if a person is no longer interested to confirm a bug still exists, the issue is closed to clear the tracker for issues people really care about.

@ilya-fedin I think it is fair to prioritize issues that are more popular. I also understand that lack of manpower and resources dictate which issues get looked at when. However, this leaves me with a few more questions. I have been subscribed to this issue for a while now. The reason I decided to comment is because when you are subscribed to an issue, Github sends you emails on every comment and the only ones I get are between the person who opened the bug report and the bot. 

So My Questions are as follows:
1. Are you able to see the number of people who are subscribed to an issue.
2. How do you determine which issues people really care about? For example: is there a way for users to thumbs up an issue or star it.
3. Are things like this https://bugs.telegram.org/c/2377 taken into consideration as well.
<br>
Thank you for taking the time to reply.

**Comment by ilya-fedin** [comment]
I don't think there's any formal system for github issues, they aren't event official, AFAIK @john-preston uses a GitHub notification bot in Telegram and from time to time looks at random issues it notifies him (I don't see a system at least).

Bugs & suggestion platform is understaffed as well AFAIK, so most issues are just ignored, but AFAIK moderators do forward some issues to @john-preston (I don't know details, it's better to ask him).

**Comment by d-air1** [comment]
@ilya-fedin Alright, I'll wait for the person you tagged to chime in. The reason I have these questions is because you stated "issues people really care about" in your original reply. This indicates popularity is the determining factor for which issues receive more focus, but as you just explained, there does not seem to be a system of popularity that we can even use. No stars, thumbs ups, subscriber count, and even the bugs & suggestions platform doesn't seem to be reliable from how you make it sound.

I just don't like reporting issues to feel like they are done in vain. Which is why it is important that everyone understands how these things work.

**Comment by ilya-fedin** [comment]
The only system of popularity here is "if no one bothers to even comment to a bot then there's no need to have the issue stay open"

**Comment by d-air1** [comment]
Yes that seems to be the case, but that isn't exactly helpful when it comes to actually getting an issue looked at. This issue has been open for over a year after all.

**Comment by ilya-fedin** [comment]
There are always way more important issues than just some unread count, sorry :(
@john-preston has a big backlog of missing features that are the priority

**Comment by d-air1** [comment]
> There are always way more important issues than just some unread count, sorry :( @john-preston has a big backlog of missing features that are the priority

I can certainly understand that. I can't even begin to argue how this is some big important issue. I'm simply interested in the process for having an issue addressed and what we as users can do to influence that.

**Comment by ilya-fedin** [comment]
I don't users can do anything :(
The only hope if community fixed bugs they're interested in, but unfortunately tdesktop has an unhelpful community...

**Comment by Bibo-Joshi** [comment]
Please keep it open

**Comment by GabrielSerealia** [comment]
Yeah been getting unreads for muted & archived channels too. Pretty annoying seeing unread count for the icon dock.

**Comment by mbrozzo** [comment]
I have this issue as well

**Comment by pingvincible** [comment]
I don't want to see message counter for archived chats, but I see them.

**Comment by isaacvonaaron** [comment]
Perhaps I could clarify a bit more on that - the purpose of using archived chat might be different for each individual, to some people - archived chats are the chats we want to browse only if we elect to go into them and read what's inside the folder.
So to users like me, and I believe many others - it would not be useful to display an unread notification or icon on the system tray because it makes us go into the app only to discover that there are no new messages of interest, rendering the unread count/notification useless to us.
I think the same could be said for other users with muted notifications. Those users might have muted the notifications because they don't have much interest in these messages and want to see them only when they elect to read them.
It would really be useful if there was a set of toggles - 
- Count unread messages
-- In muted messages
-- In archived messages

**Comment by dakamakat** [comment]
We have Settings -> Notification and Sounds -> Badge counter -> include muted chats in unread count ( when  i disable it this issue is gone for my archiwe chats and other unread and muted stuff ) is that issue stil really exist?  also a lot of relevant issues pop up on it 
#24284  
#17471  
and others

**Comment by GabrielSerealia** [comment]
Wow, actually that's exactly what I was looking for. Thanks!

**Comment by Bibo-Joshi** [comment]
I did indeed not know about this and this is helpful, however I do agree with https://github.com/telegramdesktop/tdesktop/issues/17471#issuecomment-1088864209, i.e. a distinction between archived / not-archived would be more important for me than between muted / un-muted.

**Comment by Aokromes** [comment]
> My problem is that "include muted chats in unread count" toggle apparently only accessible on Windows version of the Telegram app. I can't find it anywhere on v.10.2.4.255298. I only want to keep unread count badges for chats I care about (e.g. personal & urgent). Is this feature planned for Mac OS?

you are using https://github.com/overtake/TelegramSwift

**Comment by Aridez** [comment]
> I did indeed not know about this and this is helpful, however I do agree with [#17471 (comment)](https://github.com/telegramdesktop/tdesktop/issues/17471#issuecomment-1088864209), i.e. a distinction between archived / not-archived would be more important for me than between muted / un-muted.

Right now we have the options to "count unread messages" and "include muted chats in unread count". I agree with @Bibo-Joshi, it would be a nice feature to have a third option with "include archived chats in unread count", to me there is a difference between muting a chat and having a grey count show up and archiving a chat so it doesn't show up at all.

**Comment by 1u** [comment]
Same problem here on Android mobile. (Notifications on desktop turned off globally)   

Maybe that helps someone to narrow down the cause of the problem: For me, the bug appears when there are unread messages in a **muted chat with multiple sub topics**.  
The main chat is muted, so should be all the topics in it. But unread messages in it are still shown as unread count in the app symbol (nowhere in the app itself, btw). When I read them (open topic and scroll down) the number disappears.  
Can someone confirm?

**Comment by soredake** [comment]
Not stale.

**Comment by parametalol** [comment]
I second the opinion that muted and archived chats should have separate settings.
